src/webpack/resolveExternals.ts

```diff
@@ -1,22 +1,9 @@
 import type webpack from 'webpack';
+import nodeExternals from 'webpack-node-externals';
 
 export default (isLibrary: boolean): webpack.ExternalsElement => {
   if (isLibrary) {
-    // UMD only supports an object
-    return {
-      react: {
-        amd: 'react',
-        commonjs: 'react',
-        commonjs2: 'react',
-        root: 'React',
-      },
-      'react-dom': {
-        amd: 'react-dom',
-        commonjs: 'react-dom',
-        commonjs2: 'react-dom',
-        root: 'ReactDOM',
-      },
-    };
+    return nodeExternals();
```

Comment:
This will not work for any project. I'd rather leverage `peerDependencies` for this.

Reply:
However it will not work everywhere either so I guess the best decision will be to leave it for particular project discretion.

Reply:
I don't like to externalize everything either. Sometimes you want things externalized (React), sometimes you just want a vendor script with all of your dependencies, and sometimes you want them as part of the bundle.

Reply:
This is just for the library. The app will decide if it includes the dependency or not in the final bundle depending on the externals of the Webpack app config.

Reply:
This config worked without changing anything in the library or the app. The library bundle just emits `module.exports = require("react")` or `module.exports = require("some-other-dependency-of-the-library")`.
If `some-other-dependency-of-the-library` is a dependency of the library, webpack then adds the package to the final bundle when analyzing the line `module.exports = require("some-other-dependency-of-the-library")` (it will be converted to `module.exports = __webpack_require__("some-other-dependency-of-the-library")`)

Reply:
The improvement here is that for example `tslib` will be included only once instead of `n` times, each one for each library.
If one library uses `tslib@^1.0.0` and another uses `tslib@^2.0.0`, then in the bundle the two versions of `tslib` are going to be bundled.
If two libraries use `tslib@^2.0.0`, then only one version will be bundled. There won't be a duplicated `tslib`.